### ISSUE TITLE
Sub-command async OnExecute

### ISSUE THREAD

**Issue by rychlym** [natemcmaster/CommandLineUtils#53]
Hello,

is it a bug, that program is  taken the control before async sub-command  OnExecute is done?

a demo example:
        
```
static int Main(string[] args)
        {
            var app = new CommandLineApplication();
```

            app.HelpOption("-h|--help");
            var optionSubject = app.Option("-s|--subject <SUBJECT>", "The subject", CommandOptionType.SingleValue);

            app.Command("subCmd", cmd =>
            {
                cmd.FullName = "Sub-command";
                cmd.Description = "Demo sub-command";
                cmd.ExtendedHelpText = "An extended help text...";
                cmd.HelpOption();
                var subCmdArg = cmd.Argument("SubArg", "The argument");

                cmd.OnExecute(async () =>
                {
                    Console.WriteLine("sub-command starting ...");
                    await Task.Delay(4000);
                    Console.WriteLine("... sub-command: done - arg: " + subCmdArg.Value);
                });
            });

            app.OnExecute(async () =>
            {
                var subject = optionSubject.HasValue()
                    ? optionSubject.Value()
                    : "world";

                Console.Write($"Hello");

                // This pause here is just for indication that some awaitable operation could happens here.
                await Task.Delay(5000);
                Console.WriteLine($" {subject}!");
                return 0;
            });

            var result = app.Execute(args);
           Console.WriteLine("Press any key to end...");
            Console.ReadKey();
           return result;
         }

**Comment by rychlym** [comment]
You are right, thanks a lot for the explanation.

**Comment by natemcmaster** [comment]
Ok. This should be fixed in [2.2.0-alpha](https://www.nuget.org/packages/McMaster.Extensions.CommandLineUtils/2.2.0-alpha) which I pushed not too long ago to nuget.org. It should be available soon.
